substrafl/algorithms/pytorch/torch_fed_pca_algo.py

```diff
@@ -0,0 +1,304 @@
+import logging
+import math
+from pathlib import Path
+from typing import Any
+from typing import List
+from typing import Optional
+
+import torch
+
+from substrafl import NpIndexGenerator
+from substrafl.algorithms.pytorch import weight_manager
+from substrafl.algorithms.pytorch.torch_base_algo import TorchAlgo
+from substrafl.remote import remote_data
+from substrafl.schemas import FedAvgAveragedState
+from substrafl.schemas import FedAvgSharedState
+from substrafl.schemas import StrategyName
+
+logger = logging.getLogger(__name__)
+
+
+class TorchLinearModel(torch.nn.Module):
+    """Define linear model to encapsulate eigenvectors.
+
+    Args:
+        in_features (int): dimension of input vectors
+        out_features (int): number of dimensions to keep as part of dimensionsality
+            reduction
+        device (str): working device, cuda or cpu
+    """
+
+    def __init__(self, in_features: int, out_features: int, device: str):
+        super().__init__()
+        self.in_features = in_features
+        self.out_features = out_features
+        self.device = device
+        self.eigen_vectors = torch.nn.Linear(self.in_features, self.out_features, bias=False).to(self.device)
+        self.eigen_vectors.weight.data.to(self.device)
+
+    def forward(self, x):
+        """Perform dimensionality reduction.
+
+        Args:
+            x (torch.Tensor): inputs to map to reduced dim space.
+                shape of x is (B, D_IN)
+
+        Returns:
+            torch.Tensor: reduced dim vectors
+        """
+        eigen_vectors = self.eigen_vectors.weight.data
+        coefs = x @ eigen_vectors.T
+        return coefs @ eigen_vectors
+
+
+class TorchFedPCAAlgo(TorchAlgo):
+    """TorchFedPCAAlgo class, inheriting from TorchAlgo and designed to perform Principal Component Analysis (PCA).
```

Comment:
The docstring is really important here to. Having an example (cf torch_fed_avg) is really usefull for users. 
Same proposition as the strategy docstring. Could you write it in the easiest way for you and we will convert it to rst ?